rs/execution_environment/fuzz/src/lib.rs

```diff
@@ -59,3 +79,145 @@ pub fn fuzzer_main() {
         }
     }
 }
+
+#[cfg(target_os = "linux")]
+fn syscall_monitor<F>(name: &str, sandbox: F)
+where
+    F: Fn(),
+{
+    match unsafe { fork() } {
+        Ok(ForkResult::Child) => {
+            sandbox();
+        }
+        Ok(ForkResult::Parent { child }) => {
+            std::thread::sleep(std::time::Duration::from_secs(1));
+            let allowed_syscalls: BTreeSet<Sysno> = BTreeSet::from([
+                Sysno::mmap,
+                Sysno::mprotect,
+                Sysno::munmap,
+                Sysno::madvise,
+                Sysno::sendmsg,
+                Sysno::sigaltstack,
+                Sysno::futex,
+                Sysno::close,
+            ]);
+            loop {
+                // This code employs a manual heuristic to determine which process PID to attach to,
+                // specifically targeting the one executing the Wasm code.
+                //
+                // The child process spawns a total of 14 threads. Since PIDs are monotonically increasing
+                // and the tid are stored in a BTreeSet, they are ordered based on their creation sequence.
+                //
+                // By tracing all PIDs and analyzing the associated syscalls, we observe that the critical
+                // threads to attach to are typically among the last few, specifically [n-2] and [n-1].
```

Comment:
Perhaps we should trace all of them?

Reply:
Initially I wanted to trace only the wasm execution but it makes sense to trace all the threads. However, there was an issue with tracer getting stuck on some PIDs. I have spawned the tracers into their own thread pool which should solve this problem